common/src/main/java/io/pravega/common/util/ConfigBuilder.java

```diff
@@ -78,6 +79,22 @@ private ConfigBuilder(String namespace, Properties properties, ConfigConstructor
         return this;
     }
 
+    /**
+     * Includes the given property and its value in the builder, if the provided value is not empty.
+     *
+     * @param property The property to set.
+     * @param value    An Optional containing The value of the property. This must be of the same type as accepted by the Property.
+     * @param <V>      Type of the property.
+     * @return This instance.
+     */
+    public <V> ConfigBuilder<T> with(Property<V> property, Optional<V> value) {
```

Comment:
After further reading, it looks like passing `Optional` to a method is frowned upon. Even Intellij issues a warning for it.

Reply:
Do you have a reference for this?

Reply:
1. Open up in Intellij and look at the in-editor static analysis warnings.
2. Google for "java optional as parameters"